config/module/versions.go

```diff
@@ -21,6 +25,30 @@ func newest(versions []string, constraint string) (string, error) {
 		return "", err
 	}
 
+	// Store whether the constraint is an explicit equality that
+	// contains a metadata requirement, so we can return a specific
+	// if requested metadata version
+	var constraintMetas []string
+	var equalsConstraint bool
+	for i := range cs {
+		constraintMeta := strings.SplitAfterN(cs[i].String(), "+", 2)
+		if len(constraintMeta) > 1 {
+			constraintMetas = append(constraintMetas, constraintMeta[1])
+		}
+	}
+
+	if len(cs) == 1 {
+		equalsConstraint = explicitEqualityConstraint.MatchString(cs.String())
+	}
+
+	if (len(cs) > 1 || !equalsConstraint) && len(constraintMetas) > 0 {
+		return "", fmt.Errorf("Constraints including metadata must have explicit equality, or are otherwise too ambiguous: %s", cs.String())
```

Comment:
The semver spec calls this "build metadata", so maybe we should use that same terminology here so it's clearer what we're talking about?